### PR TITLE
Git pillar update interval

### PR DESCRIPTION
**Pull request by sathieu**
### What does this PR do?

Add support for new config option `git_pillar_update_interval` so git_pillar data can be collected less frequently than the main `loop_interval` timer allows

### What issues does this PR fix or reference?

This is based on work on #52274 by @Tanoti

### Previous Behavior

Git pillar remotes would be checked using the `loop_interval` timer value (e.g. every 60 seconds).

### New Behavior

Git pillar remotes can be checked using a different interval to the `loop_interval`. For example, if the git pillar data is not likely to change very often then a higher value such as 300 seconds could be used which would reduce the load on the git repo server by a factor of five.

### Tests written?

No

### Commits signed with GPG?

Yes

### CODE REVIEW

**Comment by sathieu** [comment]
I don't know how to write tests for this...

**Comment by sathieu** [comment]
@twangboy @dwoz @terminalmage @cachedout Please review.

**Review by DmitryKuzmenko — APPROVED** [review]

**Comment by waynew** [comment]
I'm not sure if there's any good way of automating the test, but a meaningful test would be to update the interval to something ridiculously short (1s? 3s?) and simply verify that it triggers an update the appropriate number of times within an interval.

File: salt/pillar/git_pillar.py [hunk 1]
```diff
@@ -350,6 +350,21 @@
         - __env__ https://mydomain.tld/pillar-appdata.git:
           - mountpoint: web/server/
 
+.. _git_pillar_update_interval:
+
+git_pillar_update_interval
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. versionadded:: 2020.2.0
```
Comment (dwoz) [inline salt/pillar/git_pillar.py]: This should be `Neon`

**Review by dwoz — CHANGES_REQUESTED** [review]

File: salt/pillar/git_pillar.py (hunk 1 above)
Comment (sathieu) [reply salt/pillar/git_pillar.py]: Done (also in `doc/ref/configuration/master.rst`), rebased and force-pushed.

**Comment by sathieu** [comment]
@dwoz thanks!

**Pull request merged**